chia/wallet/util/tx_config.py

```diff
@@ -84,7 +84,9 @@ def autofill(
         constants: ConsensusConstants = kwargs["constants"]
         return CoinSelectionConfig(
             min_coin_amount=uint64(0) if self.min_coin_amount is None else self.min_coin_amount,
-            max_coin_amount=uint64(constants.MAX_COIN_AMOUNT) if self.max_coin_amount is None else self.max_coin_amount,
+            max_coin_amount=uint64(constants.MAX_COIN_AMOUNT)
+            if (self.max_coin_amount is None or self.max_coin_amount == 0)
+            else self.max_coin_amount,
```

Comment:
`0` as a max coin amount is a valid option to specify so I don't think this is the place to fix this issue.  Instead, we should fix whatever is using 0 as an explicit default here.